### PR TITLE
gh-117657: Fix data races reported by TSAN on `interp->threads.main`

### PR DESCRIPTION
**Pull request by mpage**
Use relaxed loads/stores when reading/writing to this field.

This fixes races like https://gist.github.com/mpage/e07497ad8dd444a789ff306cb7996acc

* Issue: gh-117657

### CODE REVIEW

**Review by colesbury — APPROVED** [review]
This looks good to me.

Can we remove any TSan suppressions after this change? I'm merging https://github.com/python/cpython/pull/118722 now, so you might want to wait a little bit and then merge `main` into this PR.

**Comment by mpage** [comment]
Sounds good. I'll wait until you merge https://github.com/python/cpython/pull/118722 then update this PR to remove the appropriate suppressions.

**Pull request merged**
